internal/storage/postgres/treecluster/get.go

```diff
@@ -7,26 +7,55 @@ import (
 
 	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
 	"github.com/green-ecolution/green-ecolution-backend/internal/logger"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
 	sqlc "github.com/green-ecolution/green-ecolution-backend/internal/storage/postgres/_sqlc"
 	"github.com/twpayne/go-geos"
 )
 
-func (r *TreeClusterRepository) GetAll(ctx context.Context) ([]*entities.TreeCluster, error) {
+func (r *TreeClusterRepository) GetAll(ctx context.Context) ([]*entities.TreeCluster, int64, error) {
 	log := logger.GetLogger(ctx)
-	rows, err := r.store.GetAllTreeClusters(ctx)
+	page, pageOk := ctx.Value("page").(int32)
+	limit, limitOK := ctx.Value("limit").(int32)
+
+	if !pageOk || !limitOK {
+		log.Debug("pagination values not found in context")
+		return nil, 0, r.store.MapError(storage.ErrPaginationValueInvalid, sqlc.TreeCluster{})
+	}
+
+	if page <= 0 || (limit != -1 && limit <= 0) {
+		log.Debug("invalid pagination values")
+		return nil, 0, r.store.MapError(storage.ErrPaginationValueInvalid, sqlc.TreeCluster{})
+	}
+
+	totalCount, err := r.store.GetAllTreeClustersCount(ctx)
+	if err != nil {
+		log.Debug("failed to get total tree cluster count in db")
+		return nil, 0, r.store.MapError(err, sqlc.TreeCluster{})
+	}
+
+	if limit == -1 {
+		limit = int32(totalCount)
+		page = 1
+	}
+
+	rows, err := r.store.GetAllTreeClusters(ctx, &sqlc.GetAllTreeClustersParams{
+		Limit:  limit,
+		Offset: (page - 1) * limit,
+	})
+
 	if err != nil {
 		log.Debug("failed to get tree clusters in db")
```

Comment:
This may be my mistake, but can you also put the error message in the debug log?

```suggestion
		log.Debug("failed to get tree clusters in db", "error", err)
```

Reply:
Did you forget to push this? :sweat_smile: There is still the error attribute missing in the log